### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by mira1zu**
-  [DEMO LINK](https://mira1zu.github.io/react_people-table-advanced/)

### CODE REVIEW

File: src/components/PeoplePage.tsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+              : (
+                <>
+                  {people.length !== 0 ? (
+                    <h3 className="text-center">
+                      Could not find any people by
+                      this query
+                    </h3>
+                  ) : (
+                    <div className="d-flex justify-content-center">
+                      <div
```
Comment (vsupruniuk) [inline src/components/PeoplePage.tsx:186]: :harold:

**Review by vsupruniuk — APPROVED** [review]
Все супер. Якби ще зробив що б radio buttons не були обрані по дефолту,  було б взагалі ідеально
image

File: src/components/NewPerson.tsx [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+        type="submit"
+        className="btn btn-primary"
+      >
+        Submit
+      </button>
+    </form>
+  );
+};
+
+export default NewPerson;
```
Comment (Olexandr6) [inline src/components/NewPerson.tsx:346]: Use name export

File: src/components/PersonRow.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+import { useParams } from 'react-router';
+
+import classNames from 'classnames';
+
+import PersonName from './PersonName';
```
Comment (Olexandr6) [inline src/components/PersonRow.tsx:7]: You don't need blank lines between these imports
```suggestion
import React from 'react';
import { useParams } from 'react-router';
import classNames from 'classnames';
import PersonName from './PersonName';
```

File: src/components/PeoplePage.tsx [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        ...prevPeople,
+        linkPersonWithParents(newPerson, prevPeople),
+      ]);
+    });
+  };
+
+  const preparePeople = () => {
+    const preparedPeople = people.filter(person => (
+      person.name.toLowerCase()
+        .includes(appliedQuery.toLowerCase())
```
Comment (Olexandr6) [inline src/components/PeoplePage.tsx:84]: you use 3 times `appliedQuery.toLowerCase()`, let's create a variable for this expression

File: src/components/PeoplePage.tsx [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+        break;
+
+      default:
+        break;
+    }
+
+    return preparedPeople;
+  };
+
+  const preparedPeople = preparePeople();
```
Comment (Olexandr6) [inline src/components/PeoplePage.tsx:117]: `useMemo`

File: src/components/PeoplePage.tsx [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+                          width: '3rem',
+                          height: '3rem',
+                        }}
+                        role="status"
+                      >
+                        <span className="visually-hidden">Loading...</span>
+                      </div>
+                    </div>
+                  )}
+                </>
```
Comment (Olexandr6) [inline src/components/PeoplePage.tsx:198]: Let's simplify these nested conditions🥲

File: src/components/PeoplePage.tsx [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+          </div>
+        </div>
+
+        <div className="col">
+          {isFormOpen
+            ? (
+              <NewPerson
+                people={people}
+                onNewPersonSubmit={handleNewPersonSubmit}
+              />
```
Comment (Olexandr6) [inline src/components/PeoplePage.tsx:146]: Let's render this Form on a separate route:
image

**Review by Olexandr6 — CHANGES_REQUESTED** [review]
Good solution👍 Let's improve it a bit👇

**Review by Olexandr6 — APPROVED** [review]

**Pull request closed without merging**